### PR TITLE
Sync Twig templateExists behaviors

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

There are 5 places in the code that does the same check, but they are not consistent. The difficulty here is to support Twig 1, 2 & 3. I think relying on `Environment::MAJOR_VERSION` is OK. `method_exists` are useless IMO. The proof is that they are currenty missing in some of them.

Basically if Twig >= 2 -> the method `exists` exists on the loader so just call it.

For Twig 1, check `ExistsLoaderInterface`, then check for `SourceContextLoaderInterface`, then call `getSource()`.

### CODE REVIEW

**Review by yceruto — APPROVED** [review]
Why not since 3.4?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.
